### DISCUSSION TITLE
getValues from validate

### DISCUSSION THREAD

**Discussion by zalizko** [react-hook-form/react-hook-form#8065] [Q&A]
Hi there,

I start using react-hook-form and comfused - can't find the way to get form data (getValues) from validate function.
My point is to extract validation functions into separate module and reuse them. But I can't do complex validation like check other fields value inside validate function.

Any solution? It would be great if validate function will receive getValues as second parameter for example.

**Comment by Moshyfawn** [comment]
Hey, @zalizko!

What do you think about [this approach](https://codesandbox.io/s/dazzling-khayyam-wufi6d?file=/src/App.tsx)? Obviously, should be a bit more generic than I've quickly written it as an example.

Some pseudocode:
```js
const validate = (formValues) => (fieldValue) => { ... some validation logic ... }
...
const { getValues } = useForm()
...
{ validate: validate(getValues()) }
```

Maybe, can be like:
```js
const validate = (getValues) => (fieldValue) => {
  const formValues = getValues()
  ... some validation logic ...
}
...
const { getValues } = useForm()
...
{ validate: validate(getValues) }
```

**Reply by zalizko** [reply]
Thank you very much for quick reply. I did mostly the same, but looking for more elegant solution :)
